Do not reset ninja movement when picking up another powerup

Robyt3 opened the pull request:
Closes #2890.

ChillerDragon commented:
Yikes a gameplay change.

Robyt3 commented:
It reverts to the behavior of 0.6. I believe it wasn't part of vanilla gameplay anyway, because vanilla maps don't contain multiple ninja powerups.

oy commented on src/game/server/entities/character.cpp:
```diff
@@ -469,7 +465,6 @@ bool CCharacter::GiveWeapon(int Weapon, int Ammo)
 void CCharacter::GiveNinja()
 {
 	m_Ninja.m_ActivationTick = Server()->Tick();
-	m_Ninja.m_CurrentMoveTime = -1;
```
When the ninja time ends while you're in a ninja move it will automatically finish that move when you pick up a new ninja powerup in the future. Cause the value is not resetted anymore here. Probably best to reset it when the ninja ends.

Robyt3 commented:
I updated it so the ninja move is reset when ninja time runs out.

I also changed the size of the `m_apHitObjects` array from `10` to `MAX_PLAYERS` so ninja will not possibly hit the 11th-15th players twice (in the unlikely scenario that this many tees are hit, e.g. when tee collision is disabled).

I removed `m_NumObjectsHit = 0;` from the hammer code because that variable has nothing to do with the hammer.
